workers/unity/Packages/io.improbable.gdk.deploymentlauncher/Tests/Editmode/DeploymentConfig/BaseDeploymentConfigTests.cs

```diff
@@ -0,0 +1,136 @@
+using System.Linq;
+using Improbable.Gdk.DeploymentLauncher.EditmodeTests.Utils;
+using NUnit.Framework;
+
+namespace Improbable.Gdk.DeploymentLauncher.EditmodeTests
+{
+    [TestFixture]
+    public class BaseDeploymentConfigTests
+    {
+        [Test]
+        public void BaseDeploymentConfig_does_not_throw_error_when_valid()
+        {
+            Assert.IsEmpty(DeploymentConfigTestUtils.GetValidBaseConfig().GetErrors());
+        }
+
+        [TestCase("demo_dpl_final")]
+        [TestCase("demo_dpl_final2")]
+        public void BaseDeploymentConfig_does_not_throw_error_for_valid_deployment_name(string deploymentName)
+        {
+            var config = DeploymentConfigTestUtils.GetValidBaseConfig();
+            config.Name = deploymentName;
+
+            Assert.IsEmpty(config.GetErrors());
+        }
+
+        [TestCase(null, DeploymentConfigErrorTypes.NullOrEmpty)]
+        [TestCase("", DeploymentConfigErrorTypes.NullOrEmpty)]
+        [TestCase("#badname!", DeploymentConfigErrorTypes.Invalid)]
+        public void BaseDeploymentConfig_throws_error_for_invalid_deployment_name(string invalidName, string errorType)
+        {
+            var config = DeploymentConfigTestUtils.GetValidBaseConfig();
+            config.Name = invalidName;
+            var errors = config.GetErrors().ToArray();
+
+            Assert.AreEqual(errors.Length, 1);
```

Comment:
I'd use the message overload the asserts to give the context for this.

In the logs you'll end up with something like `Expected 1 Found 0`